test/tests/multiapps/initial_transfer/master.i

```diff
@@ -0,0 +1,55 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  xmin = 0
+  xmax = 1
+  ymin = 0
+  ymax = 1
+  nx = 2
+  ny = 2
+[]
+
+[Problem]
+  kernel_coverage_check = false
+[]
+
+[Variables][dummy][][]
+
+[Functions]
+  [func]
+    type = ConstantFunction
+    value = 1
+  []
+[]
+
+[Postprocessors]
+  [c]
+    type = FunctionValuePostprocessor
+    function = func
+    execute_on = initial
+    # this will force this postprocessor to be evaluated before transfer on initial
+    force_preic = true
+  []
+[]
+
+[Executioner]
+  type = Steady
+[]
+
+[MultiApps]
+  [sub]
+    type = FullSolveMultiApp
+    input_files = sub.i
+    execute_on = initial
+  [../]
+[]
+
+[Transfers]
+  [to_sub]
+    type = MultiAppPostprocessorTransfer
+    direction = to_multiapp
+    multi_app = sub
+    from_postprocessor = c
```

Comment:
Did you look at the possibility of forcing the pre_ic flag by determining if the post processor was actually used in an initial transfer? Just curious. Anytime we can do something automatically we should.

Reply:
Yaqi- can you comment on my question?

Reply:
Oh, sorry, I missed your question. 

I guess it is possible to let the transfer directly do this:
```
warehouse.update(postprocessor_to_be_transferred_on_iniital, AttribPreIC(warehouse, true));
```

Is this a possible solution? I am asking because I do not know much about the warehouse. Also since transfer is added after postprocessor, we should be able to obtain the object of the postprocessor here. A danger is that we do not have full dependency resolution here. We actually need the resolution anyway but I guess we are missing it currently. If we push this to the user side, user will be responsible on making sure things are set up properly.

Reply:
My second thought is that: not like at timesteps, where multiapp can run at `begin` or at `end`, on iniital, multiapp is always run at like at `begin` before master's initial evaluation. A better solution could be add an option `initial_end` for multiapps. That can also solve the issue in the test. I am not so sure about what will be the correct design.

This `force_preic` could be useful by itself as a user case recently showed up in our user list where he has a ic depending on a vpp, but apparently currently only uo (not postprocessor or vpp) works fine. We possibly need to enhance IC UO dependency.